### PR TITLE
Remove v2-prefix from other documentation where appropriate

### PR DESCRIPTION
**Pull request by aspidites**
Note: I'm aware of @tomjaguarpaw 's PR [here](https://github.com/haskell/cabal/pull/9087) and intend to rebase once that lands as not to duplicate their efforts. 

The general approach I took here was to search for any file that mentioned "v2-" that wasn't a test file and remove the prefix if:
- it appeared to be a recommendation for what command to run
- didn't appear to be commenting on the aliased commands

Happy to consider any cases that I've missed or revert anything that does what we don't want. My goal at the end of this is to not have cabal mention `v2` _anywhere_ it's not appropriate, whether that's in command line output, documentation, or otherwise. I'm happy to update tests if that's what's required. 

I intend to complete this work (and thus properly fill out the checkbox below) over this weekend (July 7 - July 9, 2023).

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.
* [x] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

Bonus points for added automated tests!

## QA Notes
This PR isn't intended to change any functional behavior. Visually, I would expect the following:
* [ ]  the commands echoed by make targets *should not* display any `v2-` commands
* [ ] none of the cabal dev scripts should print usage information that suggests `v2-` commands
* [ ] benchmarks should not use `v2-` commands

### CODE REVIEW

**Comment by aspidites** [comment]
I also did a brief search of issues, and found the following:
- https://github.com/haskell/cabal/issues/7979: don't intend to address this
- https://github.com/haskell/cabal/issues/6213: says it's addressed by @tomjaguarpaw 's change, but I think I recall seeing some documentation with wording that could still be addressed. If I can get some assistance with copy (happy to take a first stab myself), then I'd be happy to incorporate this, otherwise I'm ignoring.
- https://github.com/haskell/cabal/issues/6105: I'm not versed enough in nix nor new-* to know how better to phrase these things, so I'm ignoring this
- https://github.com/haskell/cabal/issues/6104: seems to now be focused on concrete issues, such as with respect to how sandboxes are no longer needed, so ignoring.

Sorry if this is noise. Just wanted to be clear about the scope I'm targeting with this humble contribution.

**Comment by aspidites** [comment]
Finished reading contributor documentation, so I'm happy to remove this from draft, though I expect to implement feedback.

File: CONTRIBUTING.md
Comment (ulysses4ever) [inline CONTRIBUTING.md:7]: I think it can be cut down further: no need to mention a) that `build` is a feature (sounds weird: a build tool like cabal is supposed to have such "feature"!), b) when it was released, c) link to the blog post.

File: doc/nix-local-build.rst
Comment (ulysses4ever) [inline doc/nix-local-build.rst:14]: I think you should strike any changes to this file altogether in the light of the Tom's PR. This will make reviewers' live easier (less files to look at). And also will ease rebasing.

File: Cabal/src/Distribution/Simple/Setup/Haddock.hs
Comment (ulysses4ever) [inline Cabal/src/Distribution/Simple/Setup/Haddock.hs:445]: I think CI fails because of this change. Basically, `v2-` has semantic meaning in the `Cabal` library.

File: doc/nix-local-build.rst
Comment (aspidites) [reply doc/nix-local-build.rst:14]: Gotcha. I was originally going to fork from his branch but didn't want to juggle multiple the remotes. Didn't even occur to me to just ignore file names he touched. Thanks

**Comment by aspidites** [comment]
Thanks for the feedback @ulysses4ever . I'll address these tomorrow

**Comment by ulysses4ever** [comment]
Thank you! Could you also take care of `.github/ISSUE_TEMPLATE/bug_report.md`? It needs removal of `v2-` and the whole sentence:

> Please use version-prefixed commands (e.g. v2-build or v1-build) to avoid ambiguity.

**Comment by aspidites** [comment]
Alright, @ulysses4ever , I've implemented all of your changes. Not sure if the practice is for the author to mark conversations as resolved, or the one commenting, so I left all of your remarks open.

File: CONTRIBUTING.md
Comment (ulysses4ever) [reply CONTRIBUTING.md:81]: I think this has ben marked as resolved but wasn't actually updated.

File: bootstrap/bootstrap.py
Comment (ulysses4ever) [reply bootstrap/bootstrap.py:461]: Same.

File: changelog.d/pr-9090
Comment (ulysses4ever) [reply changelog.d/pr-9090:5]: Same: `significance` should be removed altogether.

**Comment by ulysses4ever** [comment]
@aspidites it's getting closer, thanks! There were three comments resolved prematurely. And indeed, it's the initial commenter's job to mark comments as resolved.

**Comment by aspidites** [comment]
Ooh. I know what happened. I did a force push forgetting you had suggestions that I merged. I think there one more that I forgot to reintroduce, so I'll try and find that either later this evening or tomorrow. I'll also try and see what's causing builds to fall

**Comment by aspidites** [comment]
Alright, @ulysses4ever, other than potentially CI, this should be good to go (reintroduced your suggestions, sorry about that).

File: changelog.d/pr-9090
Comment (ulysses4ever) [inline changelog.d/pr-9090:1]: I’m sorry for one more ask, but in the end this patch turned out to be a non-user visible change, so it doesn’t need a changelog file. I figured that when I was thinking about the `packages` field: it doesn’t really belong to any package. So I propose we remove the changelog file altogether.

Comment (aspidites) [reply changelog.d/pr-9090]: No worries. I've removed this file.

Comment (ulysses4ever) [reply changelog.d/pr-9090]: Good! Approving now. We'll need to find another reviewer but don't worry about that: I'll try to ping relevant folks.

**Review by ulysses4ever — APPROVED** [review]

File: CONTRIBUTING.md
Comment (geekosaur) [inline CONTRIBUTING.md:9]: Should that be `Cabal`?

File: CONTRIBUTING.md
Comment (geekosaur) [inline CONTRIBUTING.md:15]: Same here. (Compare the next diff block.)

**Review by geekosaur — APPROVED** [review]
I wonder if there's a rebase / merge conflict in here with the other ticket as well, but I don't think there's anything to be done until it manifests.

File: CONTRIBUTING.md
Comment (aspidites) [reply CONTRIBUTING.md:9]: I doubt it. If so, I'd say that's a separate problem from using the `v2-` version of commands.

File: CONTRIBUTING.md
Comment (aspidites) [reply CONTRIBUTING.md:15]: Actually, looking at the diff below, I think lowercase is correct here. Lowercase is building the entire project, where as the below variation is building individual components. Note how there's examples for both `Cabal-tests` and `cabal-tests`.

**Comment by aspidites** [comment]
> I wonder if there's a rebase / merge conflict in here with the other ticket as well, but I don't think there's anything to be done until it manifests.

I updated with the latest version of master and don't touch the files the other (if you mean @tomjaguarpaw 's) ticket touches, so there shouldn't be conflicts.

File: CONTRIBUTING.md
Comment (ulysses4ever) [reply CONTRIBUTING.md:9]: It's good. Believe it or not, `cabal build cabal` does the right thing (builds the exe).

**Comment by ulysses4ever** [comment]
@aspidites I think @geekosaur meant another patch: https://github.com/haskell/cabal/pull/9096 so, if you could kick out bootstrap/README, that would help me down the road.

**Comment by geekosaur** [comment]
Yes, that one.

**Comment by aspidites** [comment]
Ah, I understand now. I've kicked bootstrap/README.md out.

**Review by ffaf1 — APPROVED** [review]
thanks

**Comment by aspidites** [comment]
No rush on my side. I just wanted to check and see if there is anything else I need to do on my end, or if it's just a matter of waiting until you folks are ready to push the button?

For instance, at work we use the squash and merge functionality from GitHub, but I wanted to make sure you all didn't expect me to squash commits on my end.

Just let me know, thanks!

**Comment by ulysses4ever** [comment]
@aspidites no worries, thank you so much for your patience and your contribution. This PR is in the pipeline as described in https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions and will be merged by the end of week if all goes well. If it's not the case please ping us again. Thank you!

**Comment by ulysses4ever** [comment]
@aspidites actually, there's a merge connect it seems. Do you have energy to look into it?

**Comment by ulysses4ever** [comment]
@mergify squash

**Comment by ulysses4ever** [comment]
@mergify rebase

**Comment by ulysses4ever** [comment]
@aspidites actually, should be good now

**Comment by ulysses4ever** [comment]
Terrific! :tada: :tada: :tada:

**Comment by aspidites** [comment]
Thanks, @ulysses4ever . Work has been busy so I didn't see your message about conflicts until now. Thanks for fixing it for me.

**Comment by ulysses4ever** [comment]
@aspidites no problem at all! Thank you so much for your contribution!

**Pull request merged**
